src/main/java/org/mastodon/mamut/io/importer/labelimage/LabelImageUtils.java

```diff
@@ -0,0 +1,370 @@
+					+ " is out of bounds. The available big data viewer source only contains " + numChannels + " channels." );
+		final AbstractSequenceDescription< ?, ?, ? > sequenceDescription = sharedBigDataViewerData.getSpimData().getSequenceDescription();
+		final List< TimePoint > frames = sequenceDescription.getTimePoints().getTimePointsOrdered();
+		final Source< ? extends RealType< ? > > source =
+				Cast.unchecked( sharedBigDataViewerData.getSources().get( labelChannelIndex ).getSpimSource() );
+		// NB: Use the dimensions of the first source and the first time point only without checking if they are equal in other sources and time points.
+		final VoxelDimensions voxelDimensions = sequenceDescription.getViewSetups().get( 0 ).getVoxelSize();
+		IntFunction< RandomAccessibleInterval< RealType< ? > > > imgProvider =
+				frameId -> Cast.unchecked( source.getSource( frameId, 0 ) );
+		LabelImageUtils.createSpotsFromLabelImage( imgProvider, frames, projectModel.getModel(), voxelDimensions, sigma,
+				projectModel.getContext().getService( StatusService.class ) );
+	}
+
+	/**
+	 * Scales the covariance matrix the given sigma and the voxel dimensions.
+	 * @param covariance the covariance matrix to scale.
+	 * @param sigma the standard deviation of the Gaussian distribution to use for the covariance matrix.
+	 * @param voxelDimensions the dimensions of the voxels in the image.
+	 * @throws IllegalArgumentException if the covariance matrix has not the same dimensions as the given voxelDimensions.
+	 */
+	public static void scale( final double[][] covariance, final double sigma, final VoxelDimensions voxelDimensions )
+	{
+		if ( covariance.length != voxelDimensions.numDimensions() )
+			throw new IllegalArgumentException( "Covariance matrix has wrong dimension." );
+		for ( int i = 0; i < covariance.length; i++ )
+		{
+			if ( covariance[ i ].length != voxelDimensions.numDimensions() )
+				throw new IllegalArgumentException( "Covariance matrix has wrong dimension." );
+			for ( int j = i; j < covariance.length; j++ )
+			{
+				covariance[ i ][ j ] *= Math.pow( sigma, 2 ) * voxelDimensions.dimension( i ) * voxelDimensions.dimension( j );
+				// the covariance matrix is symmetric!
+				if ( i != j )
+					covariance[ j ][ i ] = covariance[ i ][ j ];
+			}
+		}
+	}
+
+	/**
+	 * A class to hold the pixel count, mean and covariances of a label in the image.
+	 */
+	private static class Label
+	{
+		private final int value;
+
+		private int numPixels;
+
+		private final MeansVector means;
+
+		private final CovarianceMatrix covariances;
+
+		private Label( final int value, final int numDimensions )
+		{
+			this.value = value;
+			numPixels = 0;
+			means = new MeansVector( numDimensions );
+			covariances = new CovarianceMatrix( numDimensions );
+		}
+
+		private void addPixel( int[] pixel )
+		{
+			numPixels++;
+			means.addValues( pixel );
+			covariances.addValues( pixel );
```

Comment:
Indeed. I extended the Covariance class to return the mean as well.
I will keep the mean classes, since the may be useful in on their own in the future.